### PR TITLE
Add GetIdentitiesAssets batch endpoint and bump dependencies

### LINKED ISSUE #3: Aggregation API: Batch Assets balances Endpoint
## Context

The wallet app currently fetches all asset holdings (tokens AND smart contract shares) for multiple addresses via `POST https://api.qubic.li/Wallet/Assets`. This qli endpoint is being deprecated and we need an aggregation endpoint to replace it.

The existing RPC live API has single-address endpoints for assets (`GET /live/v1/assets/{identity}/owned`, `/possessed`, `/issued`), but the wallet does **not** use them — it only uses the qli batch endpoint. The qli endpoint aggregates both owned and possessed data into a single flat response, covering all asset types:

- **Tokens** (e.g., CFB, RANDOM, QFT, QWALLET, etc.) — fungible assets issued by various identities
- **Smart contract shares** (e.g., QX shares, IPO shares) — shares in smart contracts

The wallet treats all of these the same way — they are displayed together in the Assets page, grouped by `publicId + assetName + issuerIdentity`.

**Reference**: The `POST /aggregation/v1/getCurrentIpoBids` endpoint is already live and follows the same request pattern (`{ "identities": [...] }`).

---

## Current qli endpoint being replaced

```
POST https://api.qubic.li/Wallet/Assets
Content-Type: application/json
Authorization: Bearer <jwt-token>

["IDENTITY_1", "IDENTITY_2"]
```

Response (`QubicAsset[]`) — flat array, one entry per identity+asset+contract combination, including both tokens and SC shares:

```json
[
  {
    "publicId": "BZBQFLLBNCXEMGLOBHUVFTLUPLVCPQUASSILFABOFFBCADQSSUPNWLZBQEXK",
    "contractIndex": 1,
    "assetName": "CFB",
    "contractName": "QX",
    "ownedAmount": 1000,
    "possessedAmount": 1000,
    "tick": 22451873,
    "reportingNodes": ["127.0.0.1"],
    "issuerIdentity": "CFBMEMZOIDEXQAUXYYSZIURADQLAPWPMNJXQSNVQZAHYVOPYUKKJBJUCTVJL"
  },
  {
    "publicId": "BZBQFLLBNCXEMGLOBHUVFTLUPLVCPQUASSILFABOFFBCADQSSUPNWLZBQEXK",
    "contractIndex": 1,
    "assetName": "QFT",
    "contractName": "QX",
    "ownedAmount": 50,
    "possessedAmount": 50,
    "tick": 22451873,
    "reportingNodes": ["127.0.0.1"],
    "issuerIdentity": "TFUYVBXYIYBVTEMJHAJGEJOOZHJBQFVQLTBBKMEHPEVIZFXZRPEYFUWGTIWG"
  },
  {
    "publicId": "BZBQFLLBNCXEMGLOBHUVFTLUPLVCPQUASSILFABOFFBCADQSSUPNWLZBQEXK",
    "contractIndex": 1,
    "assetName": "QWALLET",
    "contractName": "QX",
    "ownedAmount": 200,
    "possessedAmount": 200,
    "tick": 22451873,
    "reportingNodes": ["127.0.0.1"],
    "issuerIdentity": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFNQC"
  },
  {
    "publicId": "EPYWDREDNLHXOFYVGQUKPHJGOMPBSLDDGZDPKVQUMFXAIQYMZGEHPZTAAWON",
    "contractIndex": 1,
    "assetName": "RANDOM",
    "contractName": "QX",
    "ownedAmount": 10,
    "possessedAmount": 10,
    "tick": 22451870,
    "reportingNodes": ["127.0.0.1"],
    "issuerIdentity": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFNQC"
  }
]
```

**Note:** The qli endpoint pre-aggregates data that on the RPC side would require calling both `/assets/{identity}/owned` and `/assets/{identity}/possessed` separately and then merging the results. The `ownedAmount` and `possessedAmount` fields come from these two different RPC sources but qli returns them combined in a single entry.

---

## What the wallet uses from this endpoint

| Field | Used | How |
|-------|------|-----|
| `publicId` | Yes | Match assets to owning seed, grouping key |
| `assetName` | Yes | Display name, grouping key, asset selection (e.g., "CFB", "QFT", "RANDOM", "QWALLET") |
| `issuerIdentity` | Yes | Displayed in UI, used in transfer payload, part of grouping key |
| `contractIndex` | Yes | Smart contract lookup, determines if asset is transferable (only QX-managed assets can be transferred) |
| `contractName` | Yes | Displayed in UI |
| `ownedAmount` | Yes | Displayed amount, max transfer validation |
| `possessedAmount` | Yes | Displayed alongside owned amount (shown as "Owned / Possessed" in the assets table) |
| `tick` | Yes | Data freshness indicator |
| `reportingNodes` | No | Not used by the wallet |

---

## Requirements

- Must return **all** asset types: tokens (CFB, QFT, RANDOM, QWALLET, etc.) and smart contract shares — the wallet does not distinguish between them
- A single identity can have multiple assets (one entry per asset per contract)
- Both `ownedAmount` and `possessedAmount` are needed — they come from different RPC sources (`/owned` vs `/possessed`) but should be returned together
- No authentication should be required (public data)

### PR DESCRIPTION
**Pull request by 0xluk**
## Summary

- Adds `POST /getIdentitiesAssets` (gRPC `GetIdentitiesAssets`) to general-service. Accepts up to 15 identities and returns a flat array of asset balances — one entry per `(public_id, asset_name, issuer_identity, contract_index)` — with both owned and possessed amounts populated side by side. Replaces the deprecated qli `POST /Wallet/Assets` used by the wallet. Closes #3.
- Per identity, owned and possessed RPCs run concurrently against qubic-http and are merged in-process; each side keeps its own `valid_for_tick`. The qli `contractName` field is intentionally omitted (the wallet maps from `contract_index` client-side).
- Bumps `go-node-connector` from `v0.17.0` to `v2.0.0` (import path change only; `Identity.FromPubKey` API is unchanged).
- Bumps all other direct deps in `general-service` and `shared` to the latest minor versions (`grpc 1.81.1`, `grpc-gateway/v2 2.29.0`, `zap 1.28.0`, `golang.org/x/sync 0.20.0`, `ardanlabs/conf/v3 3.12.0`, `go-data-publisher/status-service 1.5.0`, `archive-query-service/v2` latest, `genproto/googleapis/api` latest).
- `general-service` now consumes `shared` via `replace github.com/qubic/qubic-aggregation/shared => ../shared`.
- Mocks regenerated to match the new `ArchiveQueryServiceClient` interface (`GetEvents` → `GetEventLogs`).

## Test plan

- [x] `cd general-service && make proto-gen` regenerates Go bindings cleanly
- [x] `cd general-service && make mocks` regenerates mocks cleanly
- [x] `cd general-service && make proto-lint` passes
- [x] `cd general-service && go vet ./...` passes
- [x] `cd general-service && go build ./...` passes
- [x] `cd general-service && go test ./...` — domain, grpc, clients, integration all green
- [x] `cd shared && go build ./...` + `go test ./...` pass
- [ ] Smoke test `POST /getIdentitiesAssets` against a live qubic-http upstream with an identity that holds both tokens and SC shares

### CODE REVIEW

File: general-service/api/qubic/aggregation/general/v1/messages.proto [hunk 1]
```diff
@@ -58,3 +58,22 @@ message GetIdentitiesBalancesRequest {
 message GetIdentitiesBalancesResponse {
   repeated IdentityBalance balances = 1;
 }
+
+message AssetBalance {
+  string public_id = 1;
+  string asset_name = 2;
+  string issuer_identity = 3;
```
Comment (qubicmio) [inline general-service/api/qubic/aggregation/general/v1/messages.proto]: I would prefer asset_issuer. it's also fitting to the glossary (but issuer identity is used in some other places, too)

File: general-service/api/qubic/aggregation/general/v1/messages.proto [hunk 2]
```diff
@@ -58,3 +58,22 @@ message GetIdentitiesBalancesRequest {
...
+  uint32 contract_index = 4;
+  int64 owned_amount = 5;
```
Comment (qubicmio) [inline general-service/api/qubic/aggregation/general/v1/messages.proto]: typically we do not use 'amount' but number of shares. Maybe quantity would be possible, if you want a short name. because amount typically refers to 'qus'.

File: general-service/api/qubic/aggregation/general/v1/messages.proto [hunk 3]
```diff
@@ -58,3 +58,22 @@ message GetIdentitiesBalancesRequest {
...
+  int64 possessed_amount = 6;
```
Comment (qubicmio) [inline general-service/api/qubic/aggregation/general/v1/messages.proto]: number of shares?

File: general-service/api/qubic/aggregation/general/v1/messages.proto [hunk 4]
```diff
@@ -58,3 +58,22 @@ message GetIdentitiesBalancesRequest {
 message GetIdentitiesBalancesResponse {
   repeated IdentityBalance balances = 1;
 }
+
+message AssetBalance {
+  string public_id = 1;
+  string asset_name = 2;
+  string issuer_identity = 3;
+  uint32 contract_index = 4;
```
Comment (qubicmio) [inline general-service/api/qubic/aggregation/general/v1/messages.proto]: There are typically two contract indices. There is a `managingContractIndex` for ownership and one `managingContractIndex` for possession.

File: general-service/api/qubic/aggregation/general/v1/messages.proto [hunk 5]
```diff
@@ -58,3 +58,22 @@ message GetIdentitiesBalancesRequest {
 message GetIdentitiesBalancesResponse {
   repeated IdentityBalance balances = 1;
 }
+
+message AssetBalance {
```
Comment (qubicmio) [inline general-service/api/qubic/aggregation/general/v1/messages.proto]: I think we should divide in ownerships and possessions. I think that would make the structure easier to understand and also more conformant to the domain data model.

File: general-service/api/qubic/aggregation/general/v1/messages.proto [hunk 6]
```diff
@@ -58,3 +58,22 @@ message GetIdentitiesBalancesRequest {
...
+  string public_id = 1;
```
Comment (qubicmio) [inline general-service/api/qubic/aggregation/general/v1/messages.proto]: What is the public id? Do you mean the `identity` (according to glossary)

File: general-service/domain/assets.go [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+	for _, p := range possessed {
+		key := assetKey{publicId: p.PublicId, assetName: p.AssetName, issuerIdentity: p.IssuerIdentity, contractIndex: p.ContractIndex}
+		idx, ok := indexByKey[key]
+		if !ok {
+			balances = append(balances, AssetBalance{
+				PublicId:              p.PublicId,
+				AssetName:             p.AssetName,
+				IssuerIdentity:        p.IssuerIdentity,
+				ContractIndex:         p.ContractIndex,
```
Comment (qubicmio) [inline general-service/domain/assets.go]: This overrides the managing contract index value from above?

File: general-service/domain/model.go [hunk 8]
```diff
@@ -63,6 +63,35 @@ type IdentityBalance struct {
...
+type PossessedAsset struct {
+	PublicId       string
+	AssetName      string
+	IssuerIdentity string
+	ContractIndex  uint32
+	Amount         int64
+	Tick           uint32
+}
+
+type AssetBalance struct {
```
Comment (qubicmio) [inline general-service/domain/model.go]: I think this domain model that combines owned and possessed assets as top level fields is not a good idea.

**Review by qubicmio — CHANGES_REQUESTED** [review]
I think we have to rethink the structure of the asset balance.

1. In our domain model ownerships and possessions are quite separated and combining them could lead to unforeseen problems (what if a possession is managed by another contract as the ownership for example - no idea if that is possible). I would prefer to keep the separation and return the data in accordance to the core domain model for easier maintenance and better compatibility.

2. We already have some inconsistencies in our asset endpoints. Therefore I think we should adhere to our glossary to not widen the gap: https://github.com/qubic/qct/blob/main/docs/glossary.md

File: general-service/api/qubic/aggregation/general/v1/messages.proto (hunk 5 above)
Comment (0xluk) [reply general-service/api/qubic/aggregation/general/v1/messages.proto]: Done in b89236a — split into per-identity `ownerships` and `possessions` lists; `AssetBalance` is gone and there's no merging step anymore.

File: general-service/api/qubic/aggregation/general/v1/messages.proto (hunk 6 above)
Comment (0xluk) [reply general-service/api/qubic/aggregation/general/v1/messages.proto]: Yes — renamed to `identity` per the glossary base term, and moved it onto the per-identity wrapper (`IdentityAssets.identity`) so it isn't repeated on every row. Fixed in b89236a.

File: general-service/api/qubic/aggregation/general/v1/messages.proto (hunk 1 above)
Comment (0xluk) [reply general-service/api/qubic/aggregation/general/v1/messages.proto]: Renamed to `asset_issuer` per glossary in b89236a.

File: general-service/api/qubic/aggregation/general/v1/messages.proto (hunk 4 above)
Comment (0xluk) [reply general-service/api/qubic/aggregation/general/v1/messages.proto]: Good catch — split. Each `AssetOwnership` and `AssetPossession` now carries its own `managing_contract_index` (taken from `OwnedAssetData.ManagingContractIndex` and `PossessedAssetData.ManagingContractIndex` respectively). Added a regression test in `assets_test.go` for the case where the two indices differ. Fixed in b89236a.

File: general-service/api/qubic/aggregation/general/v1/messages.proto (hunk 2 above)
Comment (0xluk) [reply general-service/api/qubic/aggregation/general/v1/messages.proto]: Renamed to `number_of_shares` per glossary in b89236a.

File: general-service/api/qubic/aggregation/general/v1/messages.proto (hunk 3 above)
Comment (0xluk) [reply general-service/api/qubic/aggregation/general/v1/messages.proto]: Renamed to `number_of_shares` per glossary in b89236a.

File: general-service/domain/assets.go (hunk 7 above)
Comment (0xluk) [reply general-service/domain/assets.go]: That whole merge step is gone in b89236a — ownerships and possessions are returned as two independent lists, so there's nothing to override.

File: general-service/domain/model.go (hunk 8 above)
Comment (0xluk) [reply general-service/domain/model.go]: Agreed — dropped the combined `AssetBalance` type. Domain model now has separate `AssetOwnership` and `AssetPossession`, grouped per identity via `IdentityAssets{Identity, Ownerships, Possessions}`. Fixed in b89236a.

**Comment by 0xluk** [comment]
Thanks for the review @qubicmio — addressed in b89236a:

1. **Split ownerships and possessions.** Dropped the combined `AssetBalance` row. The endpoint now returns `{ assets: [{ identity, ownerships, possessions }, ...] }` with ownership and possession as independent lists per identity. Each side keeps its own `managing_contract_index` and `tick_number`, so the "possession managed by a different contract" edge case is represented faithfully. Added a regression test in `domain/assets_test.go` (`DifferentManagingContracts`).
2. **Glossary alignment.** Renamed all fields to match https://github.com/qubic/qct/blob/main/docs/glossary.md: `identity`, `asset_issuer`, `asset_name`, `managing_contract_index`, `number_of_shares`, `tick_number`. Domain types mirror the proto.

Replied inline to each individual comment.

File: general-service/api/qubic/aggregation/general/v1/service.proto [hunk 9]
```diff
@@ -57,4 +57,17 @@ service AggregationGeneralService {
...
+  rpc GetIdentitiesAssets(GetIdentitiesAssetsRequest) returns (GetIdentitiesAssetsResponse) {
+    option (gnostic.openapi.v3.operation) = {
+      summary: "Get asset balances for a list of identities."
+      description: "Allows for requesting the owned and possessed asset balances (tokens and smart contract shares) for a number of Qubic wallet identities."
+    };
+
+    option (google.api.http) = {
+      post: "/getIdentitiesAssets"
+      body: "*"
+      response_body: "assets"
```
Comment (qubicmio) [inline general-service/api/qubic/aggregation/general/v1/service.proto]: If I understand correctly we only return the list items as outermost objects without a container around them (like identityAssets = { ... }), correct? This reminds me of the issue we had in the query service where we couldn't add anymore outer objects to the response later (like pagination information). Here it is probably not necessary but I wonder, if we should keep the possibility open to add something later. I think that would be safer.

Furthermore I think the generated openapi documentation isn't correct when omitting the 'assets' property (at least I noticed this in the other endpoint). This is a separate issue but maybe worth considering.

File: general-service/README.md [hunk 10]
```diff
@@ -79,6 +79,56 @@ General-purpose aggregation service. Currently provides IPO bid transaction aggr
...
+{
+  "identities": ["IDENTITY_A", "IDENTITY_B"]
+}
+```
+
+**Response**: per-identity asset holdings split into separate ownership and possession lists, following the [glossary](https://github.com/qubic/qct/blob/main/docs/glossary.md) names. Ownership and possession may live under different managing contracts, so they're returned independently rather than merged.
+
+```json
+{
+  "assets": [
```
Comment (qubicmio) [inline general-service/README.md]: if I read the grpc response definition correctly the 'assets' is not present in the response. (see above)

**Review by qubicmio — APPROVED** [review]
lgtm
